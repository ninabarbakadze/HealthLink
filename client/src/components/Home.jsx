import React from "react";
import { Button, Container } from "@material-ui/core";
import { Redirect } from 'react-router-dom';
import { useUser } from "../context/UserContext";
import { Box } from "@mui/system";
import {useHistory} from 'react-router-dom';
import NavBar from './NavBar/NavBar';

const Home = ({authorization}) => {

    let history = useHistory();

    const {user, isDoctor} = useUser();




    if(!authorization){
        return <Redirect to="login"/>;
    }

    const visualizeUserAppointments = () => {

    }

    //lint tag in a click handler
    const createNewAppointement =  () => {
      history.push("/new");
    }

    return (
        <div>
          <NavBar />

          <Container component="main" maxWidth="md">
            <Box
              sx={{
                marginTop: 8,
                display: 'flex',
                flexDirection: 'row',
                justifyContent: 'space-between',
                alignItems: 'center',
              }}
            >

              <Button variant="contained">Visualize my Appointments</Button>
              { isDoctor ? null : <Button variant="contained" onClick={createNewAppointement}>Create a new appointment</Button>
              }

            </Box>
        </Container>
        </div>
    )
}

export default Home;
