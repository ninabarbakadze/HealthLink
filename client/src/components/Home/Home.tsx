import { FunctionComponent } from "react";
import NavBar from '../NavBar/NavBar';
import { Redirect } from 'react-router-dom';
import { useUser } from '../../context/UserContext';

interface HomeProp {
  authorization: boolean
}

 const Home: FunctionComponent<HomeProp> = ({ authorization }) => {

  

  const {user, isDoctor} = useUser();




  if(!authorization){
      return <Redirect to="login"/>;
  }

  const visualizeUserAppointments = () => {
  
  }

  const createNewAppointement =  () => {
    <Redirect to='/new' />
  }

  return (
      <div>
        <NavBar />
            <button>Visualize my Appointments</button>
            { isDoctor ? null : <button onClick={createNewAppointement}>Create a new appointment</button>
            }
      </div>
  )
};

export default Home