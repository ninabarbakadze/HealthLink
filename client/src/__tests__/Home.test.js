import  Home  from '../components/Home.jsx';
import { render } from '@testing-library/react';
// import {useUser} from '../context/UserContext';

// const mockUser = {
//   id:'string',
//   isdoctor:false,
//   name:'string',
//   age:10,
//   email:'string',
//   username:'string',
//   password:'string',
//   stripeid:'string',
//   peerid: 'string'
// }

//test if it loads
describe('home component', () => {
  test('should match the snapshot', () => {
    //  const { setUser } = useUser();
    //  setUser(mockUser);
     const { container } = render(
         <Home authorization={true} />
     );
     expect(container.firstChild).toMatchSnapshot();
   })
});


//test for visualize and create
//test logout
