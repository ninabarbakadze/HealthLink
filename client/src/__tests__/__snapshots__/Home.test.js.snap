// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home component should match the snapshot 1`] = `
<div>
  <div
    class="MuiBox-root css-i9gxme"
  >
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-hip9hq-MuiPaper-root-MuiAppBar-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1t29gy6-MuiToolbar-root"
      >
        <button
          aria-label="menu"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeLarge css-ancrnh-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="MenuIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <div
          class="MuiTypography-root MuiTypography-h6 css-1juivf6-MuiTypography-root"
        >
          News
        </div>
        <button
          class="MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorInherit MuiButtonBase-root css-1y942vo-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Logout
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </header>
  </div>
  <main
    class="MuiContainer-root MuiContainer-maxWidthMd"
  >
    <div
      class="MuiBox-root css-zwz5fz"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Visualize my Appointments
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Create a new    Appointment
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </main>
</div>
`;
